Radar chart with accessibility plugin generates "Expected length, NaN" error on null value.

#### Expected behaviour
* No console errors are generated when rendering a Radar chart with null values in the series with the accessibility module included.

#### Actual behaviour
* When creating a radar chart with the latest Highchart version, if the accessibility plugin is used and the series has null values, it will generate the following console error:

```
highcharts.js:50 Error: <rect> attribute x: Expected length, "NaN".
highcharts.js:50 Error: <rect> attribute y: Expected length, "NaN".
```

#### Live demo with steps to reproduce

The original issue shown in this example:
https://jsfiddle.net/obcs95d3/

If you change the null value to a regular value the error is gone:
https://jsfiddle.net/obcs95d3/1/

If you remove the accessibility module but still have null value, the error is also gone:
https://jsfiddle.net/obcs95d3/2/

#### Product version
<!--- Highcharts, Highstock or Highmaps plus version number -->
"highcharts": "9.0.1"

#### Affected browser(s)
* Chrome: 89.0.4389.90
* Safari:  14.0.3 (16610.4.3.1.7)

--- a/samples/unit-tests/chart/polar/demo.js
+++ b/samples/unit-tests/chart/polar/demo.js
@@ -1,6 +1,5 @@
-QUnit.test('Polar chart with no data (#5226)', function (assert) {
-    assert.expect(0);
-    Highcharts.chart('container', {
+QUnit.test('Polar chart data', function (assert) {
+    const chart = Highcharts.chart('container', {
         chart: {
             polar: true
         },
@@ -13,6 +12,24 @@ QUnit.test('Polar chart with no data (#5226)', function (assert) {
             }
         ]
     });
+
+    assert.ok(
+        true,
+        '#5226: Polar chart with no data should not throw'
+    );
+
+    chart.series[0].update({
+        data: [null]
+    });
+
+    assert.notOk(
+        Number.isNaN(chart.series[0].points[0].plotX),
+        '#15438: plotX should not be NaN with null data'
+    );
+    assert.notOk(
+        Number.isNaN(chart.series[0].points[0].plotY),
+        '#15438: plotY should not be NaN with null data'
+    );
 });
 QUnit.test(
     'Polar and categorized chart should not render extra alternate band.(#2248)',

--- a/ts/Extensions/Polar.ts
+++ b/ts/Extensions/Polar.ts
@@ -302,8 +302,7 @@ seriesProto.toXY = function (
     this: Highcharts.PolarSeries,
     point: Highcharts.PolarPoint
 ): void {
-    var xy,
-        chart = this.chart,
+    var chart = this.chart,
         xAxis = this.xAxis,
         yAxis = this.yAxis,
         plotX = point.plotX,
@@ -328,14 +327,12 @@ seriesProto.toXY = function (
         radius += yAxis.center[3] / 2;
     }
 
-    // Find the polar plotX and plotY
-    xy = inverted ? yAxis.postTranslate(plotY, radius) :
-        xAxis.postTranslate(plotX, radius);
+    // Find the polar plotX and plotY. Avoid setting plotX and plotY to NaN when
+    // plotY is undefined (#15438)
+    if (isNumber(plotY)) {
+        const xy = inverted ? yAxis.postTranslate(plotY, radius) :
+            xAxis.postTranslate(plotX, radius);
 
-    // #15438: Avoid setting plotX and plotY to NaN, we only need to check x
-    // since both x and y will be NaN if any of the postTranslate arguments are
-    // undefined or NaN
-    if (isNumber(xy.x)) {
         point.plotX = point.polarPlotX = xy.x - chart.plotLeft;
         point.plotY = point.polarPlotY = xy.y - chart.plotTop;
     }

--- a/ts/Extensions/Polar.ts
+++ b/ts/Extensions/Polar.ts
@@ -332,8 +332,13 @@ seriesProto.toXY = function (
     xy = inverted ? yAxis.postTranslate(plotY, radius) :
         xAxis.postTranslate(plotX, radius);
 
-    point.plotX = point.polarPlotX = xy.x - chart.plotLeft;
-    point.plotY = point.polarPlotY = xy.y - chart.plotTop;
+    // #15438: Avoid setting plotX and plotY to NaN, we only need to check x
+    // since both x and y will be NaN if any of the postTranslate arguments are
+    // undefined or NaN
+    if (isNumber(xy.x)) {
+        point.plotX = point.polarPlotX = xy.x - chart.plotLeft;
+        point.plotY = point.polarPlotY = xy.y - chart.plotTop;
+    }
 
     // If shared tooltip, record the angle in degrees in order to align X
     // points. Otherwise, use a standard k-d tree to get the nearest point

